src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -960,11 +960,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Return Nothing 'ConversionKind.NoConversion
             End If
 
-            Dim sourceType As TypeSymbol = source.Type
+            Dim sourceType As TypeSymbol = If(source.Kind = BoundKind.TupleLiteral,
```

Comment:
It seems strange this logic is needed for tuple literals, but similar logic isn't applied for array literals (since they have an InferredType too).